model/jpa/src/main/java/org/keycloak/models/jpa/RoleAdapter.java

```diff
@@ -214,7 +201,7 @@ public int hashCode() {
         return getId().hashCode();
     }
 
-    public static RoleEntity toRoleEntity(RoleModel model, EntityManager em) {
+    private static RoleEntity toRoleEntity(RoleModel model, EntityManager em) {
```

Comment:
It doesn't have to be static any more, thanks. The check is although still necessary. `model` could be instance of `infinispan.RoleAdapter`

Reply:
`static` is better since there is one parameter less to pass (`this`). See also Effective Java by J Bloch

Reply:
> It doesn't have to be static any more, thanks. The check is although still necessary. `model` could be instance of `infinispan.RoleAdapter`

I see it now, thanks for explanation. I thought the model can't be anything else than `jpa.RoleAdapter`.

> `static` is better since there is one parameter less to pass (`this`).

Well if it would be non-static, there are still 2 parameter because we can get rid of EntityManager, since it is part of `this`. Looking at it again, I would say it is not a big difference between static and non-static here, but maybe I am missing something. We can leave it as it is.

Reply:
You're right. The rule for `static` is still valid but in this case, if the `em` is removed, it would make no difference.

Reply:
I've left it as private non static method.